olcut-core/src/main/java/com/oracle/labs/mlrg/olcut/config/ConfigurationManager.java

```diff
@@ -1474,6 +1475,66 @@ public int getNumInstantiated() {
         return lookupAll(c);
     }
 
+    /**
+     * Looks for a single instance of a specified Configurable in the configuration. If there is
+     * no such instance, this method returns null. If there is more than one instance, an
+     * exception is thrown. lookupSingleton can optionally include any class that may be
+     * assignable (a subclass or implementation) to the provided type. Note that this
+     * could potentially allow a user to load their own class. If the security of the
+     * class you're looking up is important, declare it final to prevent code insertion.
+     *
+     * @param c The Class of component to lookup.
+     * @param allowAssignable allow types that are assignable to the given class to match
+     * @param <T> The type of the component.
+     * @return the once instance of the desired class this configuration manager knows about or null if
+     *         no such instance is present
+     * @throws PropertyException if there is more than one instance
+     */
+    @SuppressWarnings("unchecked") // Casts to T are implicitly checked as we use Class<T> to find the names.
+    public <T extends Configurable> T lookupSingleton(Class<T> c, boolean allowAssignable) throws PropertyException {
```

Comment:
Should it return an optional?

Reply:
None of the other lookup methods use an Optional and many of them can return null, so I think we should stay consistent here. If we feel up to a breaking the API in the next major release, I'd use that as the time to go through and change all of them to use Optional. Though I'm personally not crazy about that idiom.

Reply:
Fair enough. It is a little cumbersome to use Optional as the return type for something like this at the moment.